common/runtime/model.go

```diff
@@ -158,63 +158,178 @@ type OIDCConfigDTO struct {
+		if issuer.User != nil {
+			*errs = append(*errs, "issuerURL must not contain a username or password")
 		}
-		if issuer != nil && issuer.User != nil {
-			errs = append(errs, "issuerURL must not contain a username or password")
+		if len(issuer.RawQuery) > 0 {
+			*errs = append(*errs, "issuerURL must not contain a query")
 		}
-		if issuer != nil && len(issuer.RawQuery) > 0 {
-			errs = append(errs, "issuerURL must not contain a query")
+		if issuer.Scheme != "https" {
+			*errs = append(*errs, "issuerURL must have https scheme")
 		}
-		if issuer != nil && issuer.Scheme != "https" {
-			errs = append(errs, "issuerURL must have https scheme")
+	}
+}
+
+func (o *OIDCsDTO) validateIssuerURLForIndex(issuerURL string, index int, errs *[]string) {
+	issuer, err := url.Parse(issuerURL)
+	if err != nil || (issuer != nil && len(issuer.Host) == 0) {
+		*errs = append(*errs, fmt.Sprintf("issuerURL must be a valid URL for OIDC at index %d", index))
+	}
+	if issuer != nil {
+		if issuer.Fragment != "" {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must not contain a fragment for OIDC at index %d", index))
+		}
+		if issuer.User != nil {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must not contain a username or password for OIDC at index %d", index))
+		}
+		if len(issuer.RawQuery) > 0 {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must not contain a query for OIDC at index %d", index))
+		}
+		if issuer.Scheme != "https" {
+			*errs = append(*errs, fmt.Sprintf("issuerURL must have https scheme for OIDC at index %d", index))
 		}
 	}
-	if len(o.SigningAlgs) != 0 {
+}
+
+func (o *OIDCsDTO) validateSigningAlgs(signingAlgs []string, errs *[]string) {
+	if len(signingAlgs) != 0 {
 		validSigningAlgs := o.validSigningAlgsSet()
-		for _, providedAlg := range o.SigningAlgs {
+		for _, providedAlg := range signingAlgs {
 			if !validSigningAlgs[providedAlg] {
-				errs = append(errs, "signingAlgs must contain valid signing algorithm(s)")
+				*errs = append(*errs, "signingAlgs must contain valid signing algorithm(s)")
 				break
 			}
 		}
 	}
+}
 
-	if len(errs) > 0 {
-		err := fmt.Errorf(strings.Join(errs, ", "))
-		return err
+func (o *OIDCsDTO) validateSigningAlgsForIndex(signingAlgs []string, index int, errs *[]string) {
+	if len(signingAlgs) != 0 {
+		validSigningAlgs := o.validSigningAlgsSet()
+		for _, providedAlg := range signingAlgs {
+			if !validSigningAlgs[providedAlg] {
+				*errs = append(*errs, fmt.Sprintf("signingAlgs must contain valid signing algorithm(s) for OIDC at index %d", index))
+				break
+			}
+		}
+	}
+}
+
+func (o *OIDCsDTO) validateRequiredClaims(requiredClaims []string, errs *[]string) {
+	if len(requiredClaims) != 0 {
+		for _, claim := range requiredClaims {
+			if !strings.Contains(claim, "=") {
+				*errs = append(*errs, fmt.Sprintf("requiredClaims must be in claim=value format, invalid claim: %s", claim))
+				continue
+			}
+			parts := strings.SplitN(claim, "=", 2)
```

Comment:
We do not detect more than one `=`.

Reply:
> func strings.SplitN(s string, sep string, n int) []string
> SplitN slices s into substrings separated by sep and returns a slice of the substrings between those separators.
> 
> The count determines the number of substrings to return:
> 
> n > 0: at most n substrings; the last substring will be the unsplit remainder;

`key=value=junk` will be split to `key`, `value=junk`. The user may provide some hash value that contains `=` so thats why we do the split only once on the first found `=`

Reply:
My bad. It is not an issue. I did not get it properly.